Description: Build zero on x32
Author: Matthias Klose <doko@ubuntu.com>
Forwarded: http://mail.openjdk.java.net/pipermail/zero-dev/2014-January/000524.html
Last-Update: 2014-07-06

Index: b/acinclude.m4
===================================================================
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -2,10 +2,18 @@ AC_DEFUN([IT_SET_ARCH_SETTINGS],
 [
   case "${host_cpu}" in
     x86_64)
-      BUILD_ARCH_DIR=amd64
-      INSTALL_ARCH_DIR=amd64
-      JRE_ARCH_DIR=amd64
-      ARCHFLAG="-m64"
+      case "${host}" in
+        *x32)
+          BUILD_ARCH_DIR=x32
+          INSTALL_ARCH_DIR=x32
+          JRE_ARCH_DIR=x32
+          ;;
+        *)
+          BUILD_ARCH_DIR=amd64
+          INSTALL_ARCH_DIR=amd64
+          JRE_ARCH_DIR=amd64
+          ARCHFLAG="-m64"
+      esac
       ;;
     i?86)
       BUILD_ARCH_DIR=i586
@@ -720,7 +728,7 @@ AC_DEFUN_ONCE([IT_ENABLE_ZERO_BUILD],
   ZERO_LIBARCH="${INSTALL_ARCH_DIR}"
   dnl can't use AC_CHECK_SIZEOF on multilib
   case "${ZERO_LIBARCH}" in
-    arm|i386|ppc|s390|sh|sparc)
+    arm|i386|ppc|s390|sh|sparc|x32)
       ZERO_BITSPERWORD=32
       ;;
     aarch64|alpha|amd64|ia64|ppc64*|s390x|sparcv9)
