--- openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_gif.c~       2014-10-02 04:17:19.000000000 +0200
+++ openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_gif.c        2014-10-29 00:23:14.428883127 +0100
@@ -310,7 +310,10 @@
     free(pBitmapBits);
     free(pOldBitmapBits);
 
-    DGifCloseFile(gif);
+    {
+    int error = 0;
+    DGifCloseFile(gif, &error);
+    }
 
     return 1;
 }
