--- openjdk/hotspot/src/os/linux/vm/os_linux.cpp.orig	2013-11-19 15:23:59.927135248 +0100
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2013-11-19 15:24:24.261540890 +0100
@@ -389,9 +389,9 @@
  *        7: The default directories, normally /lib and /usr/lib.
  */
 #if defined(AMD64) || defined(_LP64) && (defined(SPARC) || defined(PPC) || defined(S390) || defined(AARCH64))
-#define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/usr/lib64/java:/usr/lib64:/lib64:/lib:/usr/lib"
 #else
-#define DEFAULT_LIBPATH "/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/usr/lib/java:/lib:/usr/lib"
 #endif
 
 #define EXTENSIONS_DIR  "/lib/ext"
