diff -dur openjdk.orig/hotspot/make/bsd/makefiles/vm.make openjdk/hotspot/make/bsd/makefiles/vm.make
--- openjdk.orig/hotspot/make/bsd/makefiles/vm.make	2013-10-16 16:17:08.000000000 +0200
+++ openjdk/hotspot/make/bsd/makefiles/vm.make	2013-11-22 12:35:42.049971299 +0100
@@ -27,6 +27,7 @@
 
 # Common build rules.
 MAKEFILES_DIR=$(GAMMADIR)/make/$(Platform_os_family)/makefiles
+include $(GAMMADIR)/make/defs.make
 include $(MAKEFILES_DIR)/rules.make
 include $(GAMMADIR)/make/altsrc.make
 
diff -dur openjdk.orig/hotspot/make/linux/makefiles/vm.make openjdk/hotspot/make/linux/makefiles/vm.make
--- openjdk.orig/hotspot/make/linux/makefiles/vm.make	2013-10-16 16:17:08.000000000 +0200
+++ openjdk/hotspot/make/linux/makefiles/vm.make	2013-11-22 12:35:42.049971299 +0100
@@ -27,6 +27,7 @@
 
 # Common build rules.
 MAKEFILES_DIR=$(GAMMADIR)/make/$(Platform_os_family)/makefiles
+include $(GAMMADIR)/make/defs.make
 include $(MAKEFILES_DIR)/rules.make
 include $(GAMMADIR)/make/altsrc.make
 
