diff -dur icedtea-2.4.3.orig/Makefile.am icedtea-2.4.3/Makefile.am
--- icedtea-2.4.3.orig/Makefile.am	2013-10-21 10:17:55.000000000 +0200
+++ icedtea-2.4.3/Makefile.am	2013-11-22 12:07:02.553682646 +0100
@@ -258,7 +258,8 @@
 
 # Patch list
 
-ICEDTEA_PATCHES = 
+ICEDTEA_PATCHES = \
+	patches/test_gamma.patch
 
 # Conditional patches
 
@@ -299,7 +300,6 @@
 
 if WITH_PAX
 ICEDTEA_PATCHES += \
-	patches/test_gamma.patch \
 	patches/pax-mark-rmic-java.patch
 endif
 
@@ -352,7 +353,6 @@
 endif
 
 if !WITH_PAX
-ICEDTEA_BOOT_PATCHES += patches/boot/test_gamma.patch
 endif
 
 if CP39408_JAVAH
